### ISSUE TITLE
SourceJS static compiler

### ISSUE THREAD

**Issue by operatino** [sourcejs/Source#87]
Taking delivery issues in mind, maybe it would be very useful to have a SourceJS static site generator, that will create a static copy of the engine contents. 

Static version will be limited to main functions only + all client-side JS, and will serve for next purposes:

* Demo libraries distribution
* Delivery to client
* Fast setup, without NodeJS

**Comment by operatino** [comment]
Could be done using grunt/gulp plugin separately. Approximate architecture and build process steps:

1. Run SourceJS 
2. Go through all specs listed in `file-tree.json` pulling with `curl` contents via HTTP and generating same catalog structure but with generated `index.html` as spec sources.
3. Remove plugins with dynamic apis like Clarify, Comments, Crowd-voice.
4. Move assets folder to static route, so they will be accessible with existing requirejs.

**Comment by cheshirsky** [comment]
Do you suppose this scenario can ease the process of "nodeless" static delivery?

**Comment by operatino** [comment]
Yes, this scenario is very useful in static style guide generators, and allows anybody to view components library without additional setup. We're definetly planning to do so at Backbase, for bundling component bundles together with static SourceJS instance. 

This static instance will still be quite limited though, but I believe we could achieve nice result with a small amount of efforts.

**Comment by Realetive** [comment]
What result with this idea? It's very good idea because allow to have result on GitHub Pages.

**Comment by operatino** [comment]
@Realetive, since SourceJS provides scalable, dynamic platform, that supposed to be easy to extend, we don't currently have plans to officially support "compile to static" functionality.

It is possible to get relatively cheap VPS options, like on Digital Ocean, or use free node hosting plan from Heroku.

I'll leave this for the community, so anybody interested could create a plugin that does this kind of caching. We will provide any help required, but we don't have enough resources and priority working on this inside the core team.
